File: Common-Interview-Problems/IsPalindrome.py
# Check if a given string is a palindrome. ex) 'radar' == True, 'owl' == False

def is_palindrome(string):
    start = 0
    end = len(string) - 1
    while start < end:
        if string[start] != string[end]:
            return False
        start += 1
        end -= 1
    return True

test1 = 'level'
test2 = 'leveled'
print(is_palindrome(test1))
print(is_palindrome(test2))

File: Common-Interview-Problems/isPalindrome.js
// Check if a given string is a palindrome. ex) 'radar' == True, 'owl' == False

function isPalindrome(string) {
    let start = 0
    let end = string.length - 1
    while (start < end) {
        if (string.charAt(start) !== string.charAt(end)) return false
        start++
        end--
    }
    return true
}

const test1 = 'level'
const test2 = 'leveled'
console.log(isPalindrome(test1))
console.log(isPalindrome(test2))
